snowfall/models/normalize.py

```diff
@@ -0,0 +1,75 @@
+from typing import Optional
+
+import random
+import torch
+from torch import Tensor
+from torch import nn
+
+
+class GradientNormalizeIn(nn.Module):
```

Comment:
I think this could use a train/test mode so that you don't add the noise during inference (might be also a reason for WER getting worse)

Reply:
Agreed.

Reply:
... should be possible to test `if self.training` for this, I believe nn.Module gives you that.